pkg/daemon/update_test.go

```diff
@@ -64,104 +66,184 @@ func TestUpdateOS(t *testing.T) {
 // reconcilable. Welcome to the longest unittest you've ever read.
 func TestReconcilable(t *testing.T) {
+	oldIgnCfg := ctrlcommon.NewIgnConfig()
 	// oldConfig is the current config of the fake system
 	oldConfig := &mcfgv1.MachineConfig{
 		Spec: mcfgv1.MachineConfigSpec{
-			Config: ctrlcommon.NewIgnConfig(),
+			Config: runtime.RawExtension{
+				Raw: helpers.MarshalOrDie(oldIgnCfg),
+			},
 		},
 	}
+	newIgnCfg := ctrlcommon.NewIgnConfig()
+
+	// Set improper version
+	newIgnCfg.Ignition.Version = "3.0.0"
 
 	// newConfig is the config that is being requested to apply to the system
 	newConfig := &mcfgv1.MachineConfig{
 		Spec: mcfgv1.MachineConfigSpec{
-			Config: ctrlcommon.NewIgnConfig(),
+			Config: runtime.RawExtension{
+				Raw: helpers.MarshalOrDie(newIgnCfg),
+			},
 		},
 	}
-
 	// Verify Ignition version mismatch react as expected
-	newConfig.Spec.Config.Ignition.Version = "2.0.0"
 	_, isReconcilable := Reconcilable(oldConfig, newConfig)
 	checkIrreconcilableResults(t, "Ignition", isReconcilable)
+
 	//reset to proper Ignition version
-	newConfig.Spec.Config.Ignition.Version = igntypes.MaxVersion.String()
+	newIgnCfg.Ignition.Version = igntypes.MaxVersion.String()
+	newConfig = &mcfgv1.MachineConfig{
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: runtime.RawExtension{
+				Raw: helpers.MarshalOrDie(newIgnCfg),
+			},
+		},
+	}
 	_, isReconcilable = Reconcilable(oldConfig, newConfig)
 	checkReconcilableResults(t, "Ignition", isReconcilable)
 
 	// Verify Networkd unit changes react as expected
-	oldConfig.Spec.Config.Networkd = igntypes.Networkd{}
-	newConfig.Spec.Config.Networkd = igntypes.Networkd{
+	oldIgnCfg.Networkd = igntypes.Networkd{}
+	oldConfig = &mcfgv1.MachineConfig{
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: runtime.RawExtension{
+				Raw: helpers.MarshalOrDie(oldIgnCfg),
+			},
+		},
+	}
+	newIgnCfg.Networkd = igntypes.Networkd{
 		Units: []igntypes.Networkdunit{
 			igntypes.Networkdunit{
 				Name: "test.network",
 			},
 		},
 	}
+	newConfig = &mcfgv1.MachineConfig{
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: runtime.RawExtension{
+				Raw: helpers.MarshalOrDie(newIgnCfg),
+			},
+		},
+	}
 	_, isReconcilable = Reconcilable(oldConfig, newConfig)
 	checkIrreconcilableResults(t, "Networkd", isReconcilable)
 
 	// Match Networkd
-	oldConfig.Spec.Config.Networkd = newConfig.Spec.Config.Networkd
-
+	oldIgnCfg.Networkd = newIgnCfg.Networkd
+	oldConfig = &mcfgv1.MachineConfig{
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: runtime.RawExtension{
+				Raw: helpers.MarshalOrDie(oldIgnCfg),
+			},
+		},
+	}
 	_, isReconcilable = Reconcilable(oldConfig, newConfig)
 	checkReconcilableResults(t, "Networkd", isReconcilable)
 
 	// Verify Disk changes react as expected
-	oldConfig.Spec.Config.Storage.Disks = []igntypes.Disk{
+	oldIgnCfg.Storage.Disks = []igntypes.Disk{
 		igntypes.Disk{
 			Device: "/one",
 		},
 	}
-
+	oldConfig = &mcfgv1.MachineConfig{
```

Comment:
This constructor is repeated so often. Make it a shared helper?

Or maybe even a `runReconcilable(oldIgn, newIgn)` that hides the `MachineConfig` intermediaries entirely.